addons/a11y/src/components/Report/HighlightToggle.tsx

```diff
@@ -136,42 +147,32 @@ class HighlightToggle extends Component<ToggleProps, {}> {
     this.props.addElement(payload);
   }
 
-  setTargetElementOutlineStyle(targetElement: HTMLElement, outlineStyle: string): void {
-    targetElement.style.outline = outlineStyle;
-  }
-
-  onToggle(): void {
-    for (let element of this.props.elementsToHighlight) {
-      if (element) {
-        const targetElement = getElementBySelectorPath(element.target[0]);
-        if (this.props.highlightedElementsMap.get(targetElement)) {
-          let originalOutline = this.props.highlightedElementsMap.get(targetElement).originalOutline;
-          if (
-            this.props.isToggledOn &&
-            this.props.highlightedElementsMap.get(targetElement).isHighlighted
-          ) {
-            this.higlightRuleLocation(targetElement, false);
-            this.saveElementDataToMap(targetElement, false, originalOutline, this.props.type);
-          } else if (
-            !this.props.isToggledOn &&
-            !this.props.highlightedElementsMap.get(targetElement).isHighlighted
-          ) {
-            this.higlightRuleLocation(targetElement, true);
-            this.saveElementDataToMap(targetElement, true, originalOutline, this.props.type);
-          }
-        }
+  onToggle = (): void => {
+    this.props.elementsToHighlight.forEach(element => {
+      const targetElement = getElementBySelectorPath(element.target[0]);
+      if (!this.props.highlightedElementsMap.has(targetElement)) {
+        return;
       }
-    }
-  }
+      const originalOutline = this.props.highlightedElementsMap.get(targetElement).originalOutline;
+      const { isHighlighted } = this.props.highlightedElementsMap.get(targetElement);
+      const { isToggledOn } = this.props;
+      if ((isToggledOn && isHighlighted) || (!isToggledOn && !isHighlighted)) {
```

Comment:
I like how you merged the conditions here. It's clean :thumbsup:

Reply:
Thank you. 😃